src/mono/mono/metadata/image.c

```diff
@@ -1846,34 +1846,70 @@ mono_image_loaded_by_guid (const char *guid)
 	return mono_image_loaded_by_guid_internal (guid, FALSE);
 }
 
+#ifdef ENABLE_NETCORE
+static const char *
+mono_get_image_culture (MonoImage *image)
+{
+	MonoTableInfo *t = &image->tables [MONO_TABLE_ASSEMBLY];
+	if (!t->rows)
+		return NULL;
+
+	guint32 cols [MONO_ASSEMBLY_SIZE];
+	mono_metadata_decode_row (t, 0, cols, MONO_ASSEMBLY_SIZE);
+	return mono_metadata_string_heap (image, cols [MONO_ASSEMBLY_CULTURE]);
+}
+#endif
+
+char *
+mono_image_get_name_with_culture_if_needed (MonoImage *image)
+{
+#ifndef ENABLE_NETCORE
+	return NULL;
+#else
+	if (!g_str_has_prefix (image->name, "data-") &&
```

Comment:
I don't think this should be netcore-specific. The function is more broadly useful, the netcore change in logical flow should happen at the caller.